aospy/automate.py

```diff
@@ -257,25 +255,37 @@ def _compute_or_skip_on_error(calc, compute_kwargs):
         return None
 
 
-def _exec_calcs(calcs, parallelize=False, **compute_kwargs):
+def _exec_calcs(calcs, parallelize=False, client=None, **compute_kwargs):
     """Execute the given calculations.
 
     Parameters
     ----------
     calcs : Sequence of ``aospy.Calc`` objects
     parallelize : bool, default False
         Whether to submit the calculations in parallel or not
+    client : distributed.Client or None
+        The distributed Client used if parallelize is set to True; if None
+        a distributed LocalCluster is used.
     compute_kwargs : dict of keyword arguments passed to ``Calc.compute``
 
     Returns
     -------
     A list of the values returned by each Calc object that was executed.
     """
     if parallelize:
-        pool = multiprocess.Pool()
-        return pool.map(lambda calc:
-                        _compute_or_skip_on_error(calc, compute_kwargs),
-                        calcs)
+        f = lambda calc: _compute_or_skip_on_error(calc, compute_kwargs)
+        if client is None:
+            with distributed.LocalCluster() as cluster:
+                with distributed.Client(cluster) as client:
+                    logging.info(
+                        'Connected to client on local cluster: {}'.format(client))
+                    with dask.set_options(get=client.get):
+                        return db.from_sequence(calcs).map(f).compute()
+        else:
+            logging.info('Connected to distributed client: {}'.format(client))
+            with dask.set_options(get=client.get):
+                return db.from_sequence(calcs).map(f).compute()
+
```

Comment:
You can avoid the repeated code (L280-283 and L285-287) by just encapsulating L280-283 into a helper function.